cli/src/main/java/com/devonfw/tools/ide/tool/jasypt/Jasypt.java

```diff
@@ -0,0 +1,104 @@
+package com.devonfw.tools.ide.tool.jasypt;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.tool.java.Java;
+import com.devonfw.tools.ide.tool.mvn.Mvn;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link ToolCommandlet} for <a href="http://www.jasypt.org/">Jasypt</a>, The java library which allows to add basic
+ * encryption capabilities with minimum effort.
+ */
+public class Jasypt extends LocalToolCommandlet {
+
+  private static final String USAGE_INFO = """
+      Jasypt encryption tool
+      Usage:
+       encrypt  <masterpassword>  <secret>             encrypt a secret with a master-password
+       decrypt  <masterpassword>  <secret>             decrypt an encrypted secret with a master-password
+      """;
+
+  private static final String CLASS_NAME_ENCRYPTION = "org.jasypt.intf.cli.JasyptPBEStringEncryptionCLI";
+
+  private static final String CLASS_NAME_DECRYPTION = "org.jasypt.intf.cli.JasyptPBEStringDecryptionCLI";
+
+  private static final String ALGORITHM = "algorithm=PBEWITHHMACSHA512ANDAES_256";
+
+  private static final String GENERATOR_CLASS_NAME = "ivGeneratorClassName=org.jasypt.iv.RandomIvGenerator";
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Jasypt(IdeContext context) {
+
+    super(context, "jasypt", Set.of(Tag.JAVA));
+  }
+
+  @Override
+  public void run() {
+
+    String[] args = this.arguments.asArray();
+
+    if (args.length == 0) {
+      this.context.info(USAGE_INFO);
+    } else if (args.length == 3 && args[0].equals("encrypt")) {
+      runJasypt(CLASS_NAME_ENCRYPTION, args);
+    } else if (args.length == 3 && args[0].equals("decrypt")) {
+      runJasypt(CLASS_NAME_DECRYPTION, args);
+    } else {
+      this.context.warning("Unknown arguments");
+      this.context.info(USAGE_INFO);
+    }
+  }
+
+  @Override
+  public boolean doInstall(boolean silent) {
+
+    getCommandlet(Java.class).install();
+
+    return super.doInstall(silent);
+  }
+
+  @Override
+  public void postInstall() {
+
+    super.postInstall();
+
+    if (Files.notExists(resolveJasyptJarPath())) {
+      installJasyptArtifact();
+    }
+  }
+
+  private void installJasyptArtifact() {
+
+    Mvn mvn = getCommandlet(Mvn.class);
+    mvn.install();
+    this.context.debug("installing jasypt artifact in: " + resolveJasyptJarPath());
+    mvn.runTool(null, "org.apache.maven.plugins:maven-dependency-plugin:3.1.2:get",
+        "-Dartifact=org.jasypt:jasypt:" + getInstalledVersion().toString());
+  }
+
```

Comment:
You correctly converted this from the bash code.
However, this does not make sense anymore. We have Jasypt here:
https://github.com/devonfw/ide-urls/tree/master/jasypt/jasypt

And `super..doInstall(silent)` will already install `jasypt` and also not in a hard-coded version (`3.1.2`) but in the version configurable via `JASYPT_VERSION`. So we actually do not need maven anymore here and you can kick this out (remove `installJasyptArtifact` method entirely).

IMHO you also need to override `isExtract()` to `false` as otherwise the `jasypt-*.jar` will be extracted.

```suggestion
```